editor/src/viewport_tools/shape_manipulation.rs

```diff
@@ -0,0 +1,1197 @@
+use glam::{DAffine2, DVec2};
+use graphene::{
+	color::Color,
+	layers::{
+		layer_info::LayerDataType,
+		style::{self, Fill, PathStyle, Stroke},
+	},
+	LayerId, Operation,
+};
+use kurbo::{BezPath, PathEl, PathSeg, Point, Vec2};
+use std::{collections::HashSet, ops::Index};
+use std::{collections::VecDeque, ops::IndexMut};
+
+use crate::{
+	consts::{COLOR_ACCENT, VECTOR_MANIPULATOR_ANCHOR_MARKER_SIZE},
+	document::DocumentMessageHandler,
+	message_prelude::{generate_uuid, DocumentMessage, Message},
+};
+
+// Helps push values that end in approximately half, plus or minus some floating point imprecision, towards the same side of the round() function
+const BIAS: f64 = 0.0001;
+
+// The angle threshold in radians that we should mirror handles if we are below
+const MINIMUM_MIRROR_THRESHOLD: f64 = 0.1;
+
+/* Light overview of structs in this file and hierarchy:
+
+						ShapeEditor
+						/		  \
+				VectorShape ... VectorShape  <- ShapeEditor contains many VectorShapes
+					/		         \
+			   VectorAnchor ...  VectorAnchor <- VectorShape contains many VectorAnchors
+
+
+					VectorAnchor <- Container for the anchor metadata and optional VectorControlPoints
+						/
+			[Option<VectorControlPoint>; 3] <- [0] is the anchor's draggable point (but not metadata), [1] is the handle1's draggable point, [2] is the handle2's draggable point
+			/				|				\
+		"Anchor"		"Handle1" 		 "Handle2" <- These are VectorControlPoints and the only editable / draggable "primitive"
+*/
```

Comment:
A lot of these comments can be turned into doc comments (///) to be visible in the auto generated docs